File: BC-W4/ArmyJava/src/abilities/RogueAbility.java
package abilities;

import units.Unit;
import exceptions.*;

public class RogueAbility extends Ability {
    public RogueAbility(Unit owner) {
        super(owner);
    }

    public void transform() throws InvalidTransformException {
        throw new InvalidTransformException();
    }

    public void infect(Unit enemy) throws InvalidInfectException {
        throw new InvalidInfectException();
    }
}

File: BC-W4/Army/Abilities/RogueAbility/RogueAbility.cpp
#include "RogueAbility.h"

RogueAbility::RogueAbility(Unit* unit) : Ability(unit) {};
RogueAbility::~RogueAbility() {};

void RogueAbility::castAbility() {
    throw HasNoSuchAbilityException();
};

void RogueAbility::castAbility(Unit* enemy) {
    throw HasNoSuchAbilityException();
};